src/codegen.c

```diff
@@ -447,16 +453,30 @@ static void codegen_expr(CodegenContext *ctx, Node *expr) {
 
   /// Literal expression. Only integer literals are supported for now.
   case NODE_LITERAL:
-    if (expr->literal.type != TK_NUMBER) DIAG(DIAG_SORRY, expr->source_location, "Emitting non-integer literals not supported");
+    if (expr->literal.type == TK_NUMBER) expr->ir = ir_immediate(ctx, expr->type, expr->literal.integer);
+    else if (expr->literal.type == TK_STRING) {
+      char buf[48] = {0};
+      static size_t string_literal_count = 0;
+      int len = snprintf(buf, 48, "__str_lit%zu", string_literal_count++);
+      // TODO: Two options. Since we can't currently assign to static
+      // variables with compile-time known constants, it means that
+      // strings go uninitialised... So what would really be ideal
+      // is if we could just, ya know, do that. (i.e. `.string` as
+      // directive or asciz or whathaveyou). Our other option is to
+      // emit (imm+store) pairs for every byte...
+      expr->ir = ir_create_static(ctx, expr->type, as_span(string_create(buf)));
```

Comment:
I'm all for this one; my idea was much worse. I was thinking about adding a `value` member to types that stores it's default value to be initialised to, and that can be updated when we want to initialise the type to something different. I feel like the way you are describing is A. more code but also B. way more understandable, in that the backend will only deal with IR stuff instead of having to handle literals all the way from the lexer/parser